lib/dialog-renderer/index.tsx

```diff
@@ -1,64 +1,96 @@
-import React, { Fragment } from 'react';
+import React, { Component, Fragment } from 'react';
+import { connect } from 'react-redux';
 import Modal from 'react-modal';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
+import { get } from 'lodash';
 
+import DialogTypes from '../../shared/dialog-types';
 import * as Dialogs from '../dialogs';
+import { closeDialog } from '../state/ui/actions';
+
+import * as S from '../state';
+import * as T from '../types';
+
+type StateProps = {
+  dialogs: T.DialogType[];
+};
+
+type DispatchProps = {
+  closeDialog: () => any;
+};
+
+type Props = StateProps & DispatchProps;
+
+export class DialogRenderer extends Component<Props> {
+  static displayName = 'DialogRenderer';
+
+  static propTypes = {
+    appProps: PropTypes.object.isRequired,
+    buckets: PropTypes.object,
+    themeClass: PropTypes.string,
+    isElectron: PropTypes.bool.isRequired,
+    isMacApp: PropTypes.bool.isRequired,
+  };
```

Comment:
Considering that we appear to have added these here we might as well make then _types_ instead of `propTypes` since it will remove some code from the runtime but enhance our safety on the compile-time

```ts
type OwnProps = {
  appProps: object;
  buckets: Record<'noteBucket' | 'tagBucket' | 'preferencesBucket', T.Bucket>;
  themeClass: string;
  isElectron: boolean;
  isMacApp: boolean;
}
```

We could also just make `appProps` and `buckets` type `object` which would satisfy enough. looking at these props highlights that we're passing too much down through the components here and while I don't think it would be good to muddle this PR with it, I think a follow-up PR is warranted where we extract the props.

looking at the different dialogs it seems like we need three types of props and we can probably eliminate all of them:
 - redux actions which formerly had to be passed down but which can be imported on the component that uses them (`closeDialog()` for instance)
 - Simperium buckets for directly manipulating and inspecting the Simperium data, which probably would be better handled in the Simperium middleware (such as the warning not to sign-out if there are pending changes). This very well could be one source of our headaches when occasionally we can't logout.
 - Props that aren't used. Some of the old Simperium props appear to be left in place even while updating has been moved to Redux instead.

Reply:
I toyed with removing the `closeDialog` function passing, but it isn't quite as simple as just adding it to the `dialog` component. The import dialog also sets that to null while an import is happening, and allows closing again after it completes. I wasn't sure how to handle that case without prop passing and it was seeming increasingly tangential to the purpose... do you think `dialog` should handle it, and if so, what should `import` do to temporarily disable closing?

Reply:
> The import dialog also sets that to null while an import is happening, and allows closing again after it completes.

Yeah I missed this. It's something I think will be clearer after this merges. I think that if we do what I mentioned about making the dialog choice explicit/cased then that will help because for the simple cases we can leave the prop out and let the more complicated case sit for its own work.

Reply:
Just wanted to note I didn't add the props here, I just moved them: https://github.com/Automattic/simplenote-electron/blob/develop/lib/dialog-renderer/index.tsx#L54

I turned the `DialogRenderer` into a component, I'm not sure what it was before.

Reply:
I missed them at the end of the file. would you mind changing them to the `Props` type here anyway? we're still splitting it up into the typed variants for state and it's just a few lines. no biggie if you don't want to - we can do it later

Reply:
Yep NBD